packages/core/src/factory.ts

```diff
@@ -6,8 +6,10 @@ import { pipe } from 'fp-ts/function';
 import { defaults } from 'lodash';
 import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, IPrismProxyConfig, IPrismOutput } from './types';
 import { getSemigroup, NonEmptyArray } from 'fp-ts/NonEmptyArray';
-import { DiagnosticSeverity } from '@stoplight/types';
+import { DiagnosticSeverity, IHttpRequest } from '@stoplight/types';
+import { IHttpNameValue } from "@stoplight/prism-http/src/types";
 import { identity } from 'fp-ts/function';
+import { getHttpConfigFromRequest } from '@stoplight/prism-http-server/src/getHttpConfigFromRequest';
```

Comment:
we do not want a circular dependency between core and http. could we change the call sites to `mockOrForward` to include calculation of the mocking config?